superset/commands/dataset/create.py

```diff
@@ -42,19 +42,12 @@ class CreateDatasetCommand(CreateMixin, BaseCommand):
     def __init__(self, data: dict[str, Any]):
         self._properties = data.copy()
 
+    @transaction(on_error=partial(on_error, reraises=DatasetCreateFailedError))
     def run(self) -> Model:
         self.validate()
-        try:
-            # Creates SqlaTable (Dataset)
-            dataset = DatasetDAO.create(attributes=self._properties, commit=False)
 
-            # Updates columns and metrics from the dataset
-            dataset.fetch_metadata(commit=False)
-            db.session.commit()
-        except (SQLAlchemyError, DAOCreateFailedError) as ex:
-            logger.warning(ex, exc_info=True)
```

Comment:
The `@transaction()` decorator logging is _slightly_ different, but it does ensure consistency.